pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -315,10 +334,14 @@ func Convert_v1_VirtualMachine_To_api_Domain(vm *v1.VirtualMachine, domain *Doma
 	domain.ObjectMeta.Name = vm.ObjectMeta.Name
 	domain.ObjectMeta.Namespace = vm.ObjectMeta.Namespace
 
-	// XXX Fix me properly we don't want automatic fallback to qemu
-	// We will solve this properly in https://github.com/kubevirt/kubevirt/pull/804
 	if _, err := os.Stat("/dev/kvm"); os.IsNotExist(err) {
```

Comment:
I think if a user configured config map and wants to disable hardware emulation we do not need to check the existence of `/dev/kvm`
```
if !c.AllowEmulation {
    domain.Spec.Type = "qemu"
} else if _, err := os.Stat("/dev/kvm"); os.IsNotExist(err) {
    return fmt.Errorf("hardware emulation device '/dev/kvm' not present")
} else if err != nil {
    return err
}

```

Reply:
@cynepco3hahue This is by design. the setting in the ConfigMap isn't to disable hardware emulation. the setting is to allow the use of software emulation. hardware emulation is so overwhelmingly beneficial that the idea isn't to "allow the user to disable it" so much as "allow the system to keep working even if it's not available".

In most cases hardware emulation being unavailable should almost certainly be treated as an error condition unless the user knows for sure in advance that it won't work (e.g. continuous integration environment).

Does that make sense?

Reply:
Oh, thanks for the explanation I missed that `domain.Spec.Type = "qemu"` will enable software emulation.